cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -0,0 +1,51 @@
+import * as express from 'express';
+import * as passport from 'passport';
+import {BaseUser} from '../user/BaseUser';
+import {UserSecService} from '../user/UserSec';
+
+interface Auth {
+  login(strategy: string, opts?: object): void;
+  protect(role: string, loginRoute: string): void;
+  isAuthenticated(req: express.Request): boolean;
+  getUserProfile(userId: string): void;
+  logout(req: express.Request): void;
+}
```

Comment:
That's cloud interface. Some of the methods here are from client and will need to be removed later.